pkg/azuredisk/controllerserver_v2.go

```diff
@@ -213,7 +215,12 @@ func (d *DriverV2) ControllerPublishVolume(ctx context.Context, req *csi.Control
 	}
 
 	caps := []*csi.VolumeCapability{volCap}
-	if !azureutils.IsValidVolumeCapabilities(caps) {
+	maxShares, err := azureutils.GetMaxShares(req.GetVolumeContext())
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "MaxShares value not supported")
+	}
+
+	if !azureutils.IsValidVolumeCapabilities(caps, maxShares) {
```

Comment:
Given that we are validating maxShares and VolumeCapabilities in CreateVolume, ControllerPublish and ControllerUnpublish and we are throwing the same errors in case of failures, should we create a separate method for validation which can be used by all 3?

Reply:
I am not sure it makes sense to do it for just this one method because I think it just creates an unnecessary layer of abstraction but maybe we can later combine and refactor all the verifications into a single method since we mostly do the same verification steps in these calls?

Reply:
Yeah, sorry. I had meant to suggest to combine all validations and verifications (maxshares, VolumeCapabilities etc) into a single method.

Reply:
I agree that there is a potential for more refactoring here, but I think we need to address this with a separate PR in the upstream as well. It is also the case that the types of requests are not the same for all the methods, so the checks are slightly different

Reply:
Sure, makes sense.